### DISCUSSION TITLE
What is the lowest overall latency ever achieved for Jammulus on local network? What HW/SW setup did you use to achieve that?

### DISCUSSION THREAD

**Discussion by marekrigan** [jamulussoftware/jamulus#2036] [General]
What is the lowest overall latency ever achieved for Jammulus on local network? What HW/SW setup did you use to achieve that? Is it possible to achieve as low as 5ms or even lower?

**Comment by jujudusud** [comment]
I don't have the number of millisecondes but, with a 10 years old pc running Arch Linux for the client and a Raspberry Pi 3B+ running Raspbian for the server linked to a NETGEAR switch, it is unperceptible.
Certainly under 10 or 15 ms.

**Comment by 38github** [comment]
I achieve 7 ms overall delay when using automatic buffers under Fedora (Xanmod kernel) and using 64 samples latency (Pipewire). I also get this low with using JACK and not only Pipewire. The computer is an older i3-4130 using integrated GPU and has 8 GB of RAM. The audio interface is the built in one but using a Scarlett 2i2 Gen2 gives the same result.

I also use Ardour, Yabridge/LinVST-X and different effects including guitar amp emulators, compressors, gates, etc.

The server is running on FreeBSD 12.

**Reply by marekrigan** [reply]
That's impressive! I will try that setup and will see whether I can achieve that low latency.

**Comment by gilgongo** [comment]
I've not seen anything much lower than about 10ms sustained latency on a LAN connection. 

BTW, even though I'm a drummer, I don't consider latency below about 30ms to be a significant problem really. I know some people agonise about it, but I just tend to compensate a bit, even at about 50ms in fact (although I admit that's not easy). 

I also suspect perceptions of latency in Jamulus are sometimes due to people listening to their own signal, which will mean even relatively workable amounts of delay will be exaggerated for them as they start to "drag" against other players.

**Reply by marekrigan** [reply]
That's my case. I am listening to my own signal. Will try to modify my setup a little bit. Thanks for suggestion!

**Comment by ignotus666** [comment]
@38github:
> I achieve 7 ms overall delay when using automatic buffers under Fedora (Xanmod kernel) and using 64 samples latency (Pipewire). I also get this low with using JACK and not only Pipewire. The computer is an older i3-4130 using integrated GPU and has 8 GB of RAM. The audio interface is the built in one but using a Scarlett 2i2 Gen2 gives the same result.

How can that be? I use a buffer size of 24 samples in Jack (Scarlett 2i2) and the reported overall latency in Jamulus is 8-9 ms. Jitter buffers are at 2/2 (is it possible to go lower than that without getting horrible sound?). Given the same settings, no matter what kernel or PC you use, it should result in the same reported latency - whether the system can provide stable audio is another matter. Just curious.

**Reply by 38github** [reply]
With Scarlett 2i2 Gen2 I get 8ms overall delay. 7ms is when I use a built in audio interface on the motherboard. The automatic buffers are down to 2/2.
